### PR TITLE
Validate admin guard for all settings writers

### PR DESCRIPTION
**Pull request by oliverguenther**
- [x] Turns the Settings::UpdateService into a contracted one
- [x] Test that only admins can use that service
- [x] Consolidate all admin settings under the same namespace and correct paths
- [x] Base all settings controllers on the same `Admin::SettingsController` that calls the service in a controlled manner
- [x] Adds a spec to traverse all menu items found in the admin menu as an unauthorized user, confirming they're being checked

### CODE REVIEW

File: app/contracts/settings/base_contract.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+module Settings
+  class BaseContract < ::BaseContract
+    include RequiresAdminGuard
```
Comment (ulferts) [inline app/contracts/settings/base_contract.rb]: Do we need the `BaseContract` in this case? There is only the `UpdateContract` inheriting from it.

File: app/contracts/settings/update_contract.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+module Settings
+  class UpdateContract < BaseContract
+  end
```
Comment (ulferts) [inline app/contracts/settings/update_contract.rb:34]: There are no specs for this contract.

**Review by ulferts — CHANGES_REQUESTED** [review]
Good change overall. Just minor remarks within the code.

Additionally this need rebasing/merging to resolve conflicts with `dev`.

If we invest a bit into harmonizing of the template/menu names, we could remove basically all the code within the individual settings controllers in favour of inheriting it all from a base settings controller.

**Review by ulferts — CHANGES_REQUESTED** [review]
The following menu items no longer work:

Emails -> Incoming Emails ---> triggers a redirect
Time and costs -----> Missing template error
Users & Permissions -> Avatars -----> Missing template error
Work packages ----> Clicking on it shows the right page but the submenu is not jumped into which is different from the other menu items with children
Work packages -> Settings ----> see above
System settings -> Repositories -----> Missing template error

**Comment by oliverguenther** [comment]
Thanks for spotting, test coverage really is abysmal in that area and there's been too much indirection going on before. I've added another spec that traverse all menu items not only as anon/non-authorized, but also as admin and ensure the page renders something and a menu item is selected.

All settings controllers now have distinct routes, there's a redirect at the route level only for `GET /admin/settings => /admin/settings/general` and the views are moved to be conforming to the Rails standard.

**Comment by ulferts** [comment]
#### Label missing in avatar settings

image

#### Label missing in cost settings

image

**Comment by oliverguenther** [comment]
I now use the plugin's name to show the breadcrumb (i.e., "Avatar" and "Time and Costs" in your screenshots).

**Review by ulferts — APPROVED** [review]

**Pull request merged**